### ISSUE TITLE
UI - Scrollbar: Mouse gets stuck in scrollbar

### ISSUE THREAD

**Issue by bryphe** [onivim/oni2#1359]
Video here: https://drive.google.com/file/d/1e8O9_tD6BkVTLURk04O9m4_7bX46qfBx/view

Repro: Click on slider in the File Explorer or SCM pane, and scroll up or down.

Occasionally the mouse will get 'stuck' - we're not properly un-capturing the mouse in that case.

This technically isn't a crash, but it's bad because it isn't something that is obviously recoverable.

**Comment by bryphe** [comment]
Likely related / similar root cause as https://github.com/onivim/oni2/issues/441

**Comment by leematos** [comment]
I'm not seeing this behavior on my Mac. Do you think it's platform specific?

**Comment by kinglouie** [comment]
It is not platform specific i think, i have this issue since day one with the minimap.

**Comment by bryphe** [comment]
Right, I believe this to be a timing issue in the code here: https://github.com/revery-ui/revery/blob/3794ee433c114d94e756f2d563ba051d821aeb8c/src/UI_Components/Slider.re#L135 that starts and releases mouse 'capture.

It seems like a 'race' between the state being set in the hooks, vs when we re-render the entire UI. So the difference in manifestations might be related to the render time. I'm looking at a more robust way to handle this.

**Comment by bryphe** [comment]
I spent some time debugging the Revery slider control today: https://github.com/revery-ui/revery/pull/789 

This issue should now be fixed by the switch to using `refs`, which was a more reliable way to ensure that the `isActive` variable and capture behavior is synchronized, and avoids those setter-race-conditions. I was able to reproduce this around 1 out of 10 times I tried, prior to this fix - now I'm not able to reproduce at all. 

This fix came into Onivim 2 with https://github.com/onivim/oni2/pull/1412 , so we should be good now! I'll close this, but keep me posted if anyone can reproduce

I still need to check the minimap, which has a slightly different codepath - might need to port some of those fixes over. Will use #441 to track that.
